### PR TITLE
F/front page design | improve discoverability

### PR DESCRIPTION
**Pull request by martabenthem**
Github issue: https://github.com/equinor/MAD-VSM-WEB/issues/113

Some feedback from my initial testing 
- [x] hit targets needs to be adjusted. SidebarNav buttons only works when clicking the icon when it should work for the whole button.
image
- [x] Let's change the url to `projects/mine`
- [x] Change default ordering to `Last modified`
- [x] Make compenents of reusable things for front-page.
- [x] Make heart component
- [x] Make orderby persistent between front-pages. (Maybe do it as a URL-query?)
- [x] Fix Tooltips in navigation bar

### CODE REVIEW

File: components/SideNavBar.tsx
Comment (SjoenH) [inline components/SideNavBar.tsx:22]: consider `onMouseEnter` instead

File: components/SideNavBar.tsx
Comment (SjoenH) [reply components/SideNavBar.tsx:30]: Consider removing double nesting and just use a `hover` CSS selector. Then we may remove the `isHighlighted` logic that is triggered `onMouseEnter` and `onMouseLeave`.

File: components/SideNavBar.tsx
Comment (SjoenH) [inline components/SideNavBar.tsx:14]: I think we don't need this. See my comment a bit down.

File: components/SideNavBar.tsx
Comment (SjoenH) [reply components/SideNavBar.tsx:22]: Or maybe just remove it. See next comment.

File: components/SortMenu.tsx
Comment (SjoenH) [inline components/SortMenu.tsx:5]: I would probably just use an `<option>`-tag instead. Ref https://developer.mozilla.org/en-US/docs/Web/HTML/Element/option
But this works fine. :+1:

File: pages/index.tsx
Comment (SjoenH) [inline pages/index.tsx:120]: Maybe keep these?

File: pages/projects/favourite.tsx
Comment (SjoenH) [inline pages/projects/favourite.tsx:13]: Remove out-of-date comments

Comment (SjoenH) [reply pages/projects/favourite.tsx:13]: And why do we show 16 items?

File: pages/projects/index.tsx
Comment (SjoenH) [inline pages/projects/index.tsx:54]: Is this still in use?

File: pages/projects/myprojects.tsx
Comment (SjoenH) [inline pages/projects/myprojects.tsx:24]: What about the sorting?

File: pages/projects/search.tsx
Comment (SjoenH) [inline pages/projects/search.tsx:23]: What is Q? Ref https://github.com/kettanaito/naming-cheatsheet#avoid-contractions

File: styles/globals.scss
Comment (SjoenH) [inline styles/globals.scss:21]: Does it make sense to move this to a more local scope instead of global?

File: components/SideNavBar.tsx
Comment (SjoenH) [inline components/SideNavBar.tsx:57]: We have a repeating pattern here... Consider making it a component instead.

**Review by SjoenH — CHANGES_REQUESTED** [review]
Great job. :muscle: 
Please see to my comments. I'm open for discussions. :sweat_smile:

**Comment by SjoenH** [comment]
image
Also, it would be good if you could add a message if there are no projects to show.

File: pages/projects/myprojects.tsx
Comment (SjoenH) [inline pages/projects/myprojects.tsx:24]: Remember this query is unique. So instead of using "projects" as the first key. It would make sense to call it "myProjects" to enable local caching. Also for the other pages.

**Review by SjoenH — CHANGES_REQUESTED** [review]

File: styles/globals.scss
Comment (martabenthem) [reply styles/globals.scss:21]: Using Search component from EDS instead

**Comment by SjoenH** [comment]
@martabenthem Please remind me about the status on this one... Front-end is done, but we are waiting on api user fix? Correct? @Magos

**Comment by SjoenH** [comment]
@martabenthem will get her recent changes out into dev environment. Notify @ntorsen when it is done.
@Magos will take QA down for a bit and do some database migration.

If @ntorsen approves of the changes in Dev environment we will go to QA env (after clear signal from @Magos ) and let our POs test the new front-page.

**Comment by SjoenH** [comment]
I'm building to dev env now: https://console.radix.equinor.com/applications/vsm/jobs/view/radix-pipeline-20210923103927-en8md

**Comment by SjoenH** [comment]
@equinor/mad-testers @Linedueo  Ready for testing in Dev environment: https://web-vsm-dev.radix.equinor.com/projects/search

**Comment by SjoenH** [comment]
Ping @martabenthem!
Some feedback from my initial testing 
- [ ] hit targets needs to be adjusted. SidebarNav buttons only works when clicking the icon when it should work for the whole button.
image
- [ ] Let's change the url to `projects/mine`
- [ ] Ordering looks to be the wrong

**Review by SjoenH — CHANGES_REQUESTED** [review]
We talked about some changes. See the description

File: components/Card/Card.tsx
Comment (SjoenH) [inline components/Card/Card.tsx:37]: Remember to close empty tags right away.
~~`<Heart> </Heart>`~~ ->`<Heart/>`

File: components/Heart.tsx
Comment (SjoenH) [inline components/Heart.tsx:22]: Nicer to do this in CSS if you can.

File: components/SortMenu.tsx
Comment (SjoenH) [inline components/SortMenu.tsx:48]: smells like python 🐍

**Review by SjoenH — APPROVED** [review]
Good work 💯

File: components/SortMenu.tsx
Comment (SjoenH) [reply components/SortMenu.tsx:48]: I guess if the lookup is undefined the Typography component displays nothing without a crash?

File: components/Heart.tsx
Comment (martabenthem) [reply components/Heart.tsx:22]: Mmm I've just tried but here we are rendering a different icon depending on whether the mouse is over the heart or not. If the mouse is elsewhere we render <Icon data={heart_outlined}> vs <Icon data={heart_filled}> when hovering. I'm not sure that can be coded in CSS?

File: components/SortMenu.tsx
Comment (martabenthem) [reply components/SortMenu.tsx:61]: Lol it does, now that you mention it 😆

Comment (martabenthem) [reply components/SortMenu.tsx:61]: Also I have tested with an undefined lookup and the Typography does not crash, it is simply empty.

File: components/Heart.tsx
Comment (SjoenH) [reply components/Heart.tsx:22]: 🙆‍♂️

**Comment by SjoenH** [comment]
Remember that we need to **squash** and merge with a **semantic commit message** when we are done.
This is to ensure that the generated changelog will catch this change. (and help us with versioning)

File: pages/projects/index.tsx
Comment (SjoenH) [inline pages/projects/index.tsx:16]: I think I want to fix this one before we release this...

File: pages/projects/mine.tsx
Comment (SjoenH) [inline pages/projects/mine.tsx:18]: I think I want to fix this one before we release this...

File: pages/projects/search.tsx
Comment (SjoenH) [inline pages/projects/search.tsx:16]: I think I want to fix this one before we release this...

File: pages/projects/favourite.tsx
Comment (SjoenH) [inline pages/projects/favourite.tsx:16]: I think I want to fix this one before we release this...
Lets how many to display on a given window.

**Review by SjoenH — CHANGES_REQUESTED** [review]

**Comment by SjoenH** [comment]
Let's get this out into QA. If the PO says anything about the number of cards on the screen, then we do it now, else we make a task for it for later improvement.

**Comment by SjoenH** [comment]
@martabenthem If you want to have a look at it. Posting my initial start on this here:
UseWindowSize hook in hook folder:
```TypeScript
import { useState, useEffect } from "react";

// Define general type for useWindowSize hook, which includes width and height
interface WindowSize {
  width: number | undefined;
  height: number | undefined;
}

// Hook
export function useWindowSize(): WindowSize {
  // Initialize state with undefined width/height so server and client renders match
  // Learn more here: https://joshwcomeau.com/react/the-perils-of-rehydration/
  const [windowSize, setWindowSize] = useState<WindowSize>({
    width: undefined,
    height: undefined,
  });

  useEffect(() => {
    // Handler to call on window resize
    function handleResize() {
      // Set window width/height to state
      setWindowSize({
        width: window.innerWidth,
        height: window.innerHeight,
      });
    }

    // Add event listener
    window.addEventListener("resize", handleResize);

    // Call handler right away so state gets updated with initial window size
    handleResize();

    // Remove event listener on cleanup
    return () => window.removeEventListener("resize", handleResize);
  }, []); // Empty array ensures that effect is only run on mount

  return windowSize;
}
```

```TypeScript
const { width, height } = useWindowSize();
const itemWidth = 332;
const itemHeight = 100;
const padding = 100;
const itemsPerPage =
  width && height 
    ? Math.floor((width / itemWidth) * (height / itemHeight))
    : 1;
```

Note that this would not be performant if added in the top of a component.

**Comment by SjoenH** [comment]
@martabenthem 
Let's set up a redirect for projects → processes.

### 1. Change folder-name to `processes`
### 2. Change all internal links
### 3. Setup server redirect

In `next.config.js`:
```JSON5
{
  source: '/projects/:path*',
  destination: '/processes/:path*',
  permanent: false
}
```
Ref https://nextjs.org/docs/api-reference/next.config.js/redirects

**Comment by SjoenH** [comment]
👋 @martabenthem @Magos 
@knudsvik sent me a message if we could have the URL be in singular `process` instead of plural `processes`.

Not sure about the best practices here, but it reads a bit nicer... `process/12` instead of `processes/12`.

A change like this makes the routing a bit more complicated, but still doable.

- `/projects` → `/processes`. _**(move the file out from `projects`-folder and rename it to `processes.tsx`)**_
- `/projects/something-more` → `/process/something-more`

@martabenthem let me know if you want some help in setting this up.

File: components/SideNavBar.tsx
Comment (SjoenH) [inline components/SideNavBar.tsx:22]: Should it be `your` instead of `my`?

**Review by SjoenH — APPROVED** [review]
Awaiting approval from NTOR and POs before squashing and merging. ( with a proper commit )

**Comment by SjoenH** [comment]
We are ready to prepare this for release.

**Pull request merged**
